File: app/src/main/java/com/najmi/oreamnos/SessionListActivity.java
package com.najmi.oreamnos;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.appbar.MaterialToolbar;
import com.najmi.oreamnos.model.UsageStats;
import com.najmi.oreamnos.utils.PreferencesManager;

import java.util.List;
import java.util.Locale;

/**
 * Activity for displaying all sessions in a dedicated full-screen list.
 */
public class SessionListActivity extends AppCompatActivity {

    private PreferencesManager prefsManager;
    private RecyclerView sessionsRecyclerView;
    private TextView emptySessionsText;
    private TextView sessionCountText;
    private SessionAdapter sessionAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        prefsManager = new PreferencesManager(this);
        applyTheme(prefsManager.getTheme());

        setContentView(R.layout.activity_session_list);

        // Setup toolbar
        MaterialToolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        toolbar.setNavigationOnClickListener(v -> finish());

        initViews();
        loadSessions();
    }

    private void initViews() {
        sessionsRecyclerView = findViewById(R.id.sessionsRecyclerView);
        emptySessionsText = findViewById(R.id.emptySessionsText);
        sessionCountText = findViewById(R.id.sessionCountText);

        sessionsRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        sessionAdapter = new SessionAdapter();
        sessionsRecyclerView.setAdapter(sessionAdapter);
    }

    private void loadSessions() {
        UsageStats stats = prefsManager.getUsageStats();
        List<UsageStats.SessionEntry> sessions = stats.getRecentSessions();

        int count = sessions != null ? sessions.size() : 0;
        sessionCountText.setText(String.format(Locale.US, "%d sessions", count));

        if (sessions == null || sessions.isEmpty()) {
            emptySessionsText.setVisibility(View.VISIBLE);
            sessionsRecyclerView.setVisibility(View.GONE);
        } else {
            emptySessionsText.setVisibility(View.GONE);
            sessionsRecyclerView.setVisibility(View.VISIBLE);
            sessionAdapter.setSessions(sessions);
        }
    }

    private void applyTheme(String theme) {
        int mode;
        switch (theme) {
            case PreferencesManager.THEME_LIGHT:
                mode = AppCompatDelegate.MODE_NIGHT_NO;
                break;
            case PreferencesManager.THEME_DARK:
                mode = AppCompatDelegate.MODE_NIGHT_YES;
                break;
            case PreferencesManager.THEME_SYSTEM:
            default:
                mode = AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM;
                break;
        }
        AppCompatDelegate.setDefaultNightMode(mode);
    }

    // ==================== SESSION ADAPTER ====================

    private static class SessionAdapter extends RecyclerView.Adapter<SessionAdapter.SessionViewHolder> {

        private List<UsageStats.SessionEntry> sessions;

        public void setSessions(List<UsageStats.SessionEntry> sessions) {
            this.sessions = sessions;
            notifyDataSetChanged();
        }

        @NonNull
        @Override
        public SessionViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
            View view = LayoutInflater.from(parent.getContext())
                    .inflate(R.layout.item_session_entry, parent, false);
            return new SessionViewHolder(view);
        }

        @Override
        public void onBindViewHolder(@NonNull SessionViewHolder holder, int position) {
            if (sessions == null || position >= sessions.size())
                return;
            UsageStats.SessionEntry session = sessions.get(position);
            holder.bind(session);
        }

        @Override
        public int getItemCount() {
            return sessions != null ? sessions.size() : 0;
        }

        static class SessionViewHolder extends RecyclerView.ViewHolder {
            private final View statusIndicator;
            private final TextView providerModelText;
            private final TextView timestampText;
            private final TextView tokensChip;

            SessionViewHolder(@NonNull View itemView) {
                super(itemView);
                statusIndicator = itemView.findViewById(R.id.statusIndicator);
                providerModelText = itemView.findViewById(R.id.providerModelText);
                timestampText = itemView.findViewById(R.id.timestampText);
                tokensChip = itemView.findViewById(R.id.tokensChip);
            }

            void bind(UsageStats.SessionEntry session) {
                // Status indicator color
                if (session.isSuccess()) {
                    statusIndicator.setBackgroundResource(R.drawable.circle_indicator);
                } else {
                    statusIndicator.setBackgroundColor(0xFFEA4335); // Red for failure
                }

                // Provider and model
                String provider = session.getProvider() != null ? capitalize(session.getProvider()) : "Unknown";
                String model = session.getModelName() != null ? session.getModelName() : "";

                if (!model.isEmpty()) {
                    providerModelText.setText(String.format("%s • %s", provider, model));
                } else {
                    providerModelText.setText(provider);
                }

                // Timestamp
                timestampText.setText(session.getFormattedTime());

                // Tokens
                if (session.isSuccess()) {
                    tokensChip.setText(String.format(Locale.US, "%,d", session.getTotalTokens()));
                } else {
                    tokensChip.setText("Failed");
                }
            }

            private String capitalize(String s) {
                if (s == null || s.isEmpty())
                    return s;
                return s.substring(0, 1).toUpperCase() + s.substring(1);
            }
        }
    }
}

File: app/src/main/java/com/najmi/oreamnos/SessionListActivity.kt
package com.najmi.oreamnos

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.najmi.oreamnos.model.UsageStats
import com.najmi.oreamnos.ui.theme.SocurateTheme
import com.najmi.oreamnos.ui.theme.SuccessGreen
import com.najmi.oreamnos.ui.theme.ErrorRed
import com.najmi.oreamnos.utils.PreferencesManager

/**
 * Activity for displaying all sessions in a dedicated full-screen list.
 * Converted to Jetpack Compose.
 */
class SessionListActivity : ComponentActivity() {
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        
        val prefsManager = PreferencesManager(this)
        val sessions = prefsManager.getUsageStats().getRecentSessions()
        
        setContent {
            SocurateTheme {
                SessionListScreen(
                    sessions = sessions,
                    onNavigateBack = { finish() }
                )
            }
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun SessionListScreen(
    sessions: List<UsageStats.SessionEntry>,
    onNavigateBack: () -> Unit
) {
    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Session History") },
                navigationIcon = {
                    IconButton(onClick = onNavigateBack) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                            contentDescription = "Back"
                        )
                    }
                },
                actions = {
                    Text(
                        text = "${sessions.size} sessions",
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                        modifier = Modifier.padding(end = 16.dp)
                    )
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.background
                )
            )
        },
        containerColor = MaterialTheme.colorScheme.background
    ) { paddingValues ->
        if (sessions.isEmpty()) {
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(paddingValues),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = "No sessions yet",
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    textAlign = TextAlign.Center
                )
            }
        } else {
            LazyColumn(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(paddingValues)
                    .padding(horizontal = 16.dp),
                verticalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                item { Spacer(modifier = Modifier.height(8.dp)) }
                
                items(sessions) { session ->
                    SessionItem(session = session)
                }
                
                item { Spacer(modifier = Modifier.height(16.dp)) }
            }
        }
    }
}

@Composable
fun SessionItem(session: UsageStats.SessionEntry) {
    Surface(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(12.dp),
        color = MaterialTheme.colorScheme.surface
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Status indicator
            Box(
                modifier = Modifier
                    .size(12.dp)
                    .clip(CircleShape)
                    .background(if (session.isSuccess()) SuccessGreen else ErrorRed)
            )
            
            Spacer(modifier = Modifier.width(12.dp))
            
            // Content
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = buildProviderModelText(session),
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurface
                )
                Spacer(modifier = Modifier.height(4.dp))
                Text(
                    text = session.formattedTime,
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
            
            // Token count chip
            Surface(
                shape = RoundedCornerShape(16.dp),
                color = MaterialTheme.colorScheme.surfaceVariant
            ) {
                Text(
                    text = if (session.isSuccess()) 
                        String.format("%,d", session.totalTokens) 
                    else 
                        "Failed",
                    style = MaterialTheme.typography.labelMedium,
                    color = if (session.isSuccess()) 
                        MaterialTheme.colorScheme.onSurfaceVariant 
                    else 
                        ErrorRed,
                    modifier = Modifier.padding(horizontal = 12.dp, vertical = 6.dp)
                )
            }
        }
    }
}

private fun buildProviderModelText(session: UsageStats.SessionEntry): String {
    val provider = session.provider?.replaceFirstChar { it.uppercase() } ?: "Unknown"
    val model = session.modelName
    
    return if (!model.isNullOrEmpty()) {
        "$provider • $model"
    } else {
        provider
    }
}
